### PR TITLE
Backend feature prediction

### LINKED ISSUE #253: Add Prediction Feature Backend
We need to add a rate showed on user's profile page, calculated by their past predictions. These predictions should be also showed in tradinq equipment page.

### PR DESCRIPTION
**Pull request by cemalaytekin**
Closes #253

### CODE REVIEW

File: app/backend/utils.js [hunk 1]
```diff
@@ -310,4 +318,85 @@ module.exports.findUserComments = async spec => {
...
+async function resultPredictions() {
+
+  // Get currencies
+  const currencies = await CurrentTradingEquipment.find()
+
+  var currentDay = new Date();
+  day_format = currentDay.toISOString().slice(0,10); // yyyy-mm-dd
+
+  // get today's all prediction data
+  const data = await TradingEquipmentPrediction.find().where('Date').equals(day_format)
```
Comment (gurkandemir) [inline app/backend/utils.js]: why dont we get yesterday’s all predictions?

File: app/backend/utils.js [hunk 2]
```diff
@@ -310,4 +318,85 @@ module.exports.findUserComments = async spec => {
...
+    const currency =  currencies.filter(function(c) {
+      return c.from == TradingEq;
+    });
+
+    let currentValue = currency[0].rate;
+    if(element.Result != "")
+      continue;
+
+    // check whether the value at the prediction time is higher or lower than the current value and determine prediction result accordingly
+    if(element.CurrentValue > currentValue){
```
Comment (gurkandemir) [inline app/backend/utils.js]: we should discuss about situations when currency value does not change

**Comment by gurkandemir** [comment]
Another idea might be that, we should save predictions not only with date but also with time information. In this context, our schedular may work for those which predicted exactly 24 hours ago.

File: app/backend/utils.js (hunk 2 above)
Comment (cemalaytekin) [reply app/backend/utils.js]: if the value does not change and user predicted either up or down, it eventually predicted wrong. So it is not a true prediction. So we can count this case a wrong prediction.

**Pull request merged**